cmd/kubeadm/app/cmd/phases/init/data.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"io"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+)
+
+// InitData is the interface to use for init phases.
+// The "initData" type from "cmd/init.go" must satisfy this interface.
+type InitData interface {
+	UploadCerts() bool
+	CertificateKey() string
+	SetCertificateKey(key string)
+	Cfg() *kubeadmapi.InitConfiguration
+	DryRun() bool
+	SkipTokenPrint() bool
+	IgnorePreflightErrors() sets.String
+	CertificateWriteDir() string
+	CertificateDir() string
+	KubeConfigDir() string
+	KubeConfigPath() string
+	ManifestDir() string
+	KubeletDir() string
+	ExternalCA() bool
+	OutputWriter() io.Writer
+	Client() (clientset.Interface, error)
+	Tokens() []string
+}
+
+// a package local type for testing purposes.
+type testInitData struct{}
```

Comment:
What about moving this into `data_test.go`?

Reply:
ok, will do.